testing/dht_network.go

```diff
@@ -56,8 +56,10 @@ func NewTestDHTNetwork(ctx context.Context, count int) *TestDHTNetwork {
 			}
 		}
 	}
-	// to give time for the DHT to update its peer table
-	time.Sleep(10 * time.Millisecond)
+	err := WaitForPeerTableToUpdate(ctx, dhts, time.Minute)
+	if err != nil {
+		panic(err)
```

Comment:
If we keep waiting for a minute and still some node hasn't updated its peers table, the tests won't be guaranteed to work. 
Thus, it's either panicing or passing a `*testing.T` and require no error. I opted for the former because it's easier.